src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +155,85 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
 
             var templateContext = new TemplateContext
             {
-                OriginalTemplate = JObject.Parse(template),
+                OriginalTemplate = JObject.Parse(rootTemplate),
                 ExpandedTemplate = templatejObject,
                 IsMainTemplate = true,
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
                 IsBicep = isBicep,
-                SourceMap = sourceMap
+                SourceMap = sourceMap,
+                PathPrefix = pathPrefix
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
-                evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                if (this.powerShellRuleEngine != null)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
-                    {
-                        evalsToNotValidate.Add(eval);
-                    }
+                    this.logger?.LogDebug("Running PowerShell rule engine");
+                    evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
-                var uniqueResults = new Dictionary<(string, int), IEvaluation>();
-                foreach (var eval in evalsToValidate)
+               
+                // Recursively handle nested templates 
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(populatedTemplate);
```

Comment:
Make sure you're *only* using `JObject.Parse`.  Remove `dynamic` everywhere.

Reply:
Removed dynamic everywhere